src/components/messenger/list/rewards-tooltip/index.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+
+import { TooltipPopup } from '../../../tooltip-popup/tooltip-popup';
+
+export interface Properties {
+  meowPreviousDayInUSD: string;
+
+  onClose: () => void;
+}
+
+interface State {
+  isTooltipOpen: boolean;
+}
+
+export class RewardsTooltip extends React.Component<Properties, State> {
+  state = {
+    isTooltipOpen: true,
+  };
+
+  componentDidMount(): void {
+    setTimeout(() => {
+      this.closeTooltip();
+    }, 3000);
```

Comment:
This is dangerous. What if the component is already destroyed when the timer runs out?

Maybe we should just have this in the saga. It seems a little weird that we have multiple ways of determining whether the tooltip should be visible or not.

Reply:
okay, removed `isTooltipOpen`, and directly used `rewards.showRewardsInTooltip` prop. We're calling the saga method here, also tested that even if the component is destroyed (i.e user clicks on the `x` icon before the tooltip auto-hides), then also this code will be triggered. In the saga itself we're checking first if the rewards tooltip is closed or not.

Reply:
> Maybe we should just have this in the saga.

Had a doubt about this one, since if we trigger auto close from the "saga", i'm not sure if there would be a time lag.. (since the moment we set the `showRewards` in saga to `true`, and the actual moment the component, i.e, the rewards tooltip which renders might be different., right?)

So, i called `this.props.onClose` from the frontend (after the component is mounted)

Reply:
> When we call put the state is updated and things are rerendered right away

Thanks, didn't knew about this. Updated. Added tests as well.